agent/engine/docker_task_engine_test.go

```diff
@@ -181,11 +184,15 @@ func TestBatchContainerHappyPath(t *testing.T) {
 		},
 	}
 
-	if cont := <-contEvents; cont.Status != api.ContainerStopped {
-		t.Fatal("Expected container to stop first")
-		assert.Equal(t, *cont.ExitCode, 0, "Exit code should be present")
-	}
-	assert.Equal(t, (<-taskEvents).Status, api.TaskStopped, "Task is not in STOPPED state")
+	event = <-stateChangeEvents
+	assert.Equal(t, event.(api.ContainerStateChange).Status, api.ContainerStopped, "Expected container to be STOPPED")
```

Comment:
(multiple occurrences) The order here is backwards, it should be `assert.Equal(t, expected, actual, "message")`.